packages/charts/src/components/tooltip/components/tooltip_table.tsx

```diff
@@ -71,22 +67,18 @@ export const TooltipTable = <D extends BaseDatum = Datum, SI extends SeriesIdent
       </div>
     );
   }
-
   const columns = props.columns.filter(({ hidden }) => {
     return !(typeof hidden === 'boolean' ? hidden : hidden?.(props.items) ?? false);
   });
 
-  const { items, pinned = false, onSelect, selected = [] } = { ...rest, ...props };
+  const { items, pinned = false, onSelect, selected = [] } = { ...props };
```

Comment:
😱

Reply:
I don't even know what it means... I completely forgot what is this. Am I correct if I fix it this way?
```suggestion
  const { items, pinned = false, onSelect, selected = [] } = props;
```

Reply:
I read this wrong initially thinking it was a useless destructure.

I think this dual spread is still needed unless `props` contains `items`, `pinned`, `onSelect` and `selected`, which in this case it does but they can be undefined.

The previous logic was spreading the `rest` values from the `useTooltipContext` as defaults if they were not defined in the `props`.

So I'd say revert this to the old way and add back the `useTooltipContext` above.

```suggestion
  const { items, pinned = false, onSelect, selected = [] } = { ...rest, ...props };
```

Reply:
done in https://github.com/elastic/elastic-charts/pull/1915/commits/d5f88e642c4240d918102d6b095472953295814d
Only `selected` and `pinned` are available in the `tooltipContext`. `onSelect` is not in the context, `items` will be overwritten by props anyway.